src/actions.ml

```diff
@@ -540,16 +668,31 @@ let update_pr ~bot_info pr_info ~gitlab_mapping ~github_mapping
     (fun () -> GitHub_mutations.add_rebase_label pr_info.issue.issue ~bot_info)
     |> Lwt.async ;
     (* Add fail status check *)
-    GitHub_mutations.send_status_check
-      ~repo_full_name:
-        (f "%s/%s" pr_info.issue.issue.owner pr_info.issue.issue.repo)
-      ~commit:pr_info.head.sha ~state:"error" ~url:""
-      ~context:"GitLab CI pipeline (pull request)"
-      ~description:
-        "Pipeline did not run on GitLab CI because PR has conflicts with base \
-         branch."
-      ~bot_info
-    |> Lwt_result.ok )
+    match bot_info.github_token with
+    | ACCESS_TOKEN _t ->
+        GitHub_mutations.send_status_check
+          ~repo_full_name:
+            (f "%s/%s" pr_info.issue.issue.owner pr_info.issue.issue.repo)
+          ~commit:pr_info.head.sha ~state:"error" ~url:""
+          ~context:"GitLab CI pipeline (pull request)"
+          ~description:
+            "Pipeline did not run on GitLab CI because PR has conflicts with \
+             base branch."
+          ~bot_info
+        |> Lwt_result.ok
+    | INSTALL_TOKEN _t -> (
+        GitHub_queries.get_repository_id ~bot_info
+          ~owner:pr_info.issue.issue.owner ~repo:pr_info.issue.issue.repo
+        >>= function
+        | repo_id ->
```

Comment:
Why don't you distinguish between `Ok` and `Error` here? Does the code even compile?

BTW, in this case it might be possible to get the repo id from the webhook payload and not have to do a query to get it (no need to take care of this optimization if you don't want to / don't have time).

Reply:
The bind function from Lwt_result is used instead of the one from Lwt, due to the local open of `Lwt_result.Infix`.